lib/que/sql.rb

```diff
@@ -84,6 +84,55 @@ module Que
       LIMIT 1
     },
 
+    # instead of keeping this worker in a exclusivee queue mode let it take work
+    # from ANY other queues with work availble.
```

Comment:
I feel we should constraint / prioritise which queues get worked. 

A Scenario I have is if "queue-a" has "we need to work on this right now" then "queue-b" is "this is important but it can wait a bit" and "queue-c" is "whatever, as long as it gets done". 

I have a worker on "queue-a" and if that is empty I would want "queue-b" to be handled next rather than "queue-c" as "queue-b" has more important items.
Also say I have a worker where I really don't want it spending any time on "queue-c" so I can process "queue-a" / "queue-b" as soon as work is available. 

With the current approach we'd end up potentially diverting resources away from where we can them e.g. we're spending time hanging out in "queue-c" when "queue-b" has more important work

Reply:
Agree in general, one consideration I'd add is we'll actually take the highest priority work from each queue so assuming we've used the same orders of magnitude we'd actually be working those in order.

Do think we can resolve most of these issues by simply constraining the queues (which should also help with the other caveats I've mentioned).

Reply:
> assuming we've used the same orders of magnitude we'd actually be working those in order

I would question if we did use the same order of magnitude across different queues as historically we've only ever been prioritising within a queue and not thinking cross-queue. 

Do agree that constraining is likely "good enough" vs expressing priority in the queue list.